File: Easy/Longest Distinct characters in string/longest-distinct-characters-in-string.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;
int longestSubstrDistinctChars (string S);
int main()
{
    int t; cin >> t;
    while (t--)
    {
        string S; cin >> S;

        cout << longestSubstrDistinctChars (S) << endl;
    }
}

// Contributed By: Pranay Bansal

// } Driver Code Ends


int longestSubstrDistinctChars (string s)
{
    vector<int> mpp(256, -1);

        int left = 0, right = 0;
        int n = s.size();
        int len = 0;
        while(right < n)
        {
            if(mpp[s[right]] != -1) left = max(mpp[s[right]] + 1, left);

            mpp[s[right]] = right;

            len = max(len, right - left + 1);
            right++;
        }
        return len;
}

File: Easy/Longest Distinct characters in string/longest-distinct-characters-in-string.java
//{ Driver Code Starts
//Initial Template for Java


import java.io.*;
import java.util.*;
class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            String S = read.readLine();
            
            Solution ob = new Solution();
            System.out.println(ob.longestSubstrDistinctChars(S));
        }
    }
}

// } Driver Code Ends


//User function Template for Java

class Solution{
    static int longestSubstrDistinctChars(String S){
        // code here
        int count=0,max=0,i=0,j;
        String s="";
        while(i<S.length())
        {
            j=i;
            while(j<S.length())
            {
                if(s.contains(Character.toString(S.charAt(j))))
                {
                    s="";
                    max=Math.max(count,max);
                    count=0;
                    break;
                }
                else{
                    s+=S.charAt(j);
                    count++;
                    j++;
                }
                
            }
            i+=1;
        }
        max=Math.max(max,count);
        return max;
    }
}